MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/labs/IndoorMapActivity.java

```diff
@@ -0,0 +1,227 @@
+        boundingBox.add(Position.fromCoordinates(-77.03791, 38.89811));
+        boundingBox.add(Position.fromCoordinates(-77.03532, 38.89811));
+        boundingBox.add(Position.fromCoordinates(-77.03532, 38.89708));
+
+        mapboxMap.setOnCameraChangeListener(new MapboxMap.OnCameraChangeListener() {
+          @Override
+          public void onCameraChange(CameraPosition position) {
+
+            if (position.zoom > 16) {
+              try {
+                if (TurfJoins.inside(Position.fromCoordinates(
+                  position.target.getLongitude(),
+                  position.target.getLatitude()),
+                  boundingBox)) {
+                  if (!levelButtonsShowing) {
+                    showLevelButton();
+                  }
+                } else {
+                  if (levelButtonsShowing) {
+                    hideLevelButton();
+                  }
+                }
+              } catch (TurfException turfException) {
+                turfException.printStackTrace();
+              }
+            } else if (levelButtonsShowing) {
+              hideLevelButton();
+            }
+          }
+        });
+
+        String geojsonString = loadJsonFromAsset("white_house_lvl_0.geojson");
+        if (geojsonString != null) {
+          indoorBuildingSource = new GeoJsonSource("indoor-building", geojsonString);
+          mapboxMap.addSource(indoorBuildingSource);
+
+          // Add the building layers since we know zoom levels in range
+          loadBuildingLayer();
+        }
+      }
+    });
+
+    Button buttonSecondLevel = (Button) findViewById(R.id.second_level_button);
+    buttonSecondLevel.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        indoorBuildingSource.setGeoJson(loadJsonFromAsset("white_house_lvl_1.geojson"));
+      }
+    });
+
+    Button buttonGroundLevel = (Button) findViewById(R.id.ground_level_button);
+    buttonGroundLevel.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        indoorBuildingSource.setGeoJson(loadJsonFromAsset("white_house_lvl_0.geojson"));
+      }
+    });
+  }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  protected void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    mapView.onDestroy();
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  protected void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  private void hideLevelButton() {
+    // When the user moves away from our bounding box region or zooms out far enough the floor level
+    // buttons are faded out and hidden.
+    levelButtonsShowing = false;
```

Comment:
This boolean has the same state as the actual visibility of `levelButtons` right? That makes this a bit redundant .